add security indicator for no leaked credentials

francoto opened the pull request:
Add indicator based on the softwareTool [GitLeaks](https://github.com/gitleaks/gitleaks) mentioned in #8

francoto commented:
Also add a second indicator on presence of critical vulnerability

dgarijo commented:
@francoto Thanks!!
I have simplified the template. Do you mind adapting your PR? It's essentially the same, but removing most of the prefixes. See: https://github.com/EVERSE-ResearchSoftware/indicators/blob/main/indicators/codemeta_completeness.json

dgarijo commented on indicators/no_critical_vulnerability.json:
```diff
@@ -0,0 +1,31 @@
+        "@id": "https://www.bestpractices.dev/en/criteria/0#0.vulnerabilities_critical_fixed",
+        "name": "OpenSSF Best practice No Leaked Credentials"
+    },
+    "schema:contact-point": {
```
```suggestion
    "contact": {
```

dgarijo commented on indicators/no_critical_vulnerability.json:
```diff
@@ -0,0 +1,31 @@
+        "schema:name": "Tom François",
+        "schema.email": "tom.francois@lapp.in2p3.fr"
+    },
+    "rs:hasQualityDimension": {
```
```suggestion
    "qualityDimension": {
```

dgarijo commented on indicators/no_critical_vulnerability.json:
```diff
@@ -0,0 +1,31 @@
+    "@context": {
+        "schema": "http://schema.org/",
+        "rs": "https://w3id.org/everse/rsqi#"
+    },
```
```suggestion
        "@context": "https://w3id.org/everse/rsqi#",
```

dgarijo replied:
The suggestion is to replace lines 2-5 with it

dgarijo commented on indicators/no_critical_vulnerability.json:
```diff
@@ -0,0 +1,31 @@
+        "schema": "http://schema.org/",
+        "rs": "https://w3id.org/everse/rsqi#"
+    },
+    "schema:name": "noCriticalVulnerability",
```
```suggestion
    "name": "noCriticalVulnerability",
```

dgarijo commented on indicators/no_critical_vulnerability.json:
```diff
@@ -0,0 +1,31 @@
+        "rs": "https://w3id.org/everse/rsqi#"
+    },
+    "schema:name": "noCriticalVulnerability",
+    "schema:description": "Checks if reported critical vulnerabilities has been fixed",
```
```suggestion
    "description": "Checks if reported critical vulnerabilities has been fixed",
```

dgarijo commented on indicators/no_critical_vulnerability.json:
```diff
@@ -0,0 +1,31 @@
+    },
+    "schema:name": "noCriticalVulnerability",
+    "schema:description": "Checks if reported critical vulnerabilities has been fixed",
+    "schema:keywords": ["security", "critical", "vulnerability"],
```
```suggestion
    "keywords": ["security", "critical", "vulnerability"],
```

dgarijo commented on indicators/no_critical_vulnerability.json:
```diff
@@ -0,0 +1,31 @@
+    "schema:name": "noCriticalVulnerability",
+    "schema:description": "Checks if reported critical vulnerabilities has been fixed",
+    "schema:keywords": ["security", "critical", "vulnerability"],
+    "schema:url": "",
```
```suggestion
```

dgarijo commented on indicators/no_critical_vulnerability.json:
```diff
@@ -0,0 +1,31 @@
+    "schema:description": "Checks if reported critical vulnerabilities has been fixed",
+    "schema:keywords": ["security", "critical", "vulnerability"],
+    "schema:url": "",
+    "schema:identifier": {"@id": "Se42"},
```
```suggestion
    "identifier": {"@id": "Se42"},
```

dgarijo commented on indicators/no_critical_vulnerability.json:
```diff
@@ -0,0 +1,31 @@
+    "schema:keywords": ["security", "critical", "vulnerability"],
+    "schema:url": "",
+    "schema:identifier": {"@id": "Se42"},
+    "schema:status": "Active",
```
```suggestion
    "status": "Active",
```

dgarijo commented on indicators/no_critical_vulnerability.json:
```diff
@@ -0,0 +1,31 @@
+    "schema:url": "",
+    "schema:identifier": {"@id": "Se42"},
+    "schema:status": "Active",
+    "schema:version": "1.0.0",
```
```suggestion
    "version": "1.0.0",
```

dgarijo commented on indicators/no_critical_vulnerability.json:
```diff
@@ -0,0 +1,31 @@
+    "schema:version": "1.0.0",
+    "schema:author": {
+        "@type": "schema:Organisation",
+        "schema:name": "OpenSSF"
```
```suggestion
        "name": "OpenSSF"
```

dgarijo commented on indicators/no_critical_vulnerability.json:
```diff
@@ -0,0 +1,31 @@
+        "@type": "schema:Organisation",
+        "schema:name": "OpenSSF"
+    },
+    "schema:source": {
```
```suggestion
    "source": {
```

dgarijo commented on indicators/no_critical_vulnerability.json:
```diff
@@ -0,0 +1,31 @@
+    },
+    "schema:contact-point": {
+        "@type": "schema:Person",
+        "schema:name": "Tom François",
```
```suggestion
        "name": "Tom François",
```

dgarijo commented on indicators/no_critical_vulnerability.json:
```diff
@@ -0,0 +1,31 @@
+    "schema:contact-point": {
+        "@type": "schema:Person",
+        "schema:name": "Tom François",
+        "schema.email": "tom.francois@lapp.in2p3.fr"
```
```suggestion
        "schema:email": "tom.francois@lapp.in2p3.fr"
```

dgarijo commented on indicators/no_critical_vulnerability.json:
```diff
@@ -0,0 +1,31 @@
+        "@id": "rs:Security",
+        "@type": "@id"
+    },
+    "schema:dateCreated": "03-04-2025"
```
```suggestion
    "created": "03-04-2025"
```

dgarijo commented on indicators/noleakcredential.json:
```diff
@@ -0,0 +1,31 @@
+{
```
please see the comments in the previous file

dgarijo reviewed (changes requested):
Some requests. Also, could you please use a consistent naming scheme for the indicators?
I suggest renaming `noleakcredential.json` to `no_leak_credential.json`.
The name you gave the file will be its identifier.

francoto commented:
Thank you Daniel, I will update my PR tomorrow.

francoto commented on indicators/no_leaked_credential.json:
```diff
@@ -0,0 +1,25 @@
+        "name": "Tom François",
+        "schema:email": "tom.francois@lapp.in2p3.fr"
+    },
+    "qualityDimension": {"@id": "https://w3id.org/everse/i/dimensions/security"},
```
I tried to replicate the qualityDimension that codemeta_completeness is using (even if the link is actually not working, yet ?)

dgarijo replied:
I think this is correct. Next, we should add the dimension. Since this is one of the root ones, I will add it, or you may too in a new PR.

francoto commented:
I updated both indicators following your suggestions and renamed the id as mentioned in your email.
I'm not sure about the qualityDImension id though, please check them out.

dgarijo commented on indicators/no_critical_vulnerability.json:
```diff
@@ -0,0 +1,24 @@
+    "status": "Active",
+    "version": "1.0.0",
+    "author": {
+        "@type": "schema:Organisation",
```
```suggestion
        "@type": "schema:Organization",
```
Organisation does not exist in schema.org

dgarijo commented on indicators/no_critical_vulnerability.json:
```diff
@@ -0,0 +1,24 @@
+    "version": "1.0.0",
+    "author": {
+        "@type": "schema:Organisation",
+        "name": "OpenSSF"
```
shall we add the URL of the organization? I think it makes sense

dgarijo commented on indicators/no_leaked_credential.json:
```diff
@@ -0,0 +1,25 @@
+    "status": "Active",
+    "version": "1.0.0",
+    "author": {
+        "@type": "schema:Organisation",
```
```suggestion
        "@type": "schema:Organization",
```

dgarijo commented on indicators/no_leaked_credential.json:
```diff
@@ -0,0 +1,25 @@
+    "version": "1.0.0",
+    "author": {
+        "@type": "schema:Organisation",
+        "name": "OpenSSF"
```
I would add url (same comment as above)

dgarijo commented on indicators/no_critical_vulnerability.json:
```diff
@@ -0,0 +1,24 @@
+        "schema:email": "tom.francois@lapp.in2p3.fr"
+    },
+    "qualityDimension": {"@id": "https://w3id.org/everse/i/dimensions/security"},
+    "created": "03-04-2025"
```
This indicator was not created in 2025, no? Do we know when was it created?
It may be up for discussion if created is the date of addition in everse, or the creation of the indicator. I would interpret it the same as author (author of the indicator). If not known, I would not add it

dgarijo reviewed (changes requested):
Small changes requests

francoto commented on indicators/no_leaked_credential.json:
```diff
@@ -0,0 +1,25 @@
+        "schema:email": "tom.francois@lapp.in2p3.fr"
+    },
+    "qualityDimension": {"@id": "https://w3id.org/everse/i/dimensions/security"},
+    "created": "03-04-2025"
```
Ok I will remove it then as I don't know when OpenSSF come up with this indicator.

francoto commented:
applied changes :
- fix organization typo
- remove `created` key as the date of the creation of the indicator is not explicit
- add url to the organization

dgarijo commented on indicators/no_leaked_credential.json:
```diff
@@ -0,0 +1,25 @@
+        "@id": "https://www.bestpractices.dev/en/criteria/0#0.no_leaked_credentials",
+        "name": "OpenSSF Best practice No Leaked Credentials"
+    },
+    "contact-point": {
```
```suggestion
    "contact": {
```

dgarijo commented:
I would have the name as "no leaked credentials", because "noLeakedCredentials" is not very human readable, but I am merging
